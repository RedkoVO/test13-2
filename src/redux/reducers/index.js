import { combineReducers } from 'redux'
import { reducer as form } from 'redux-form'
import { APP_RESET } from '../actions/types'
import auth from './auth'
import balance from './balance'
import config from './config'
import history from './history'

const appReducer = combineReducers({
  form,
  auth,
  balance,
  config,
  history
})

export default (state, action) => {
  if (action.type === APP_RESET) {
    state = undefined
  }

  return appReducer(state, action)
}
