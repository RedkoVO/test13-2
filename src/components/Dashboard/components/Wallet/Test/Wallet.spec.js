import * as React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'

import Main from '../'

const allBalanceResultMock = {
  ADA: {
    code: 262,
    balance: '0',
    lastBlock:
      '0000000000000000000000000000000000000000000000000000000000000000'
  },
  USD: {
    currency: 'USD',
    balance: '999918599998995600000000000000000000',
    lastBlock:
      '0000000000000000000000000000000000000000000000000000000000000000'
  }
}

const Wrapper = reduxForm({ form: 'test' })(({ children }) => children)
const store = createStore(a => a, {})

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Wrapper>
          <Main
            onSubmit={() => {}}
            onCopyKey={() => {}}
            handleChangeBalance={() => {}}
            allBalanceResult={allBalanceResultMock}
            curencySelectValue={{ currency: 'DCB', code: 0 }}
            isDisabledButton={false}
          />
        </Wrapper>
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
