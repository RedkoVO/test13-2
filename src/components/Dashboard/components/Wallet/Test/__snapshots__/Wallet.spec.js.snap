// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Wallet-root-1"
>
  <div
    className="Wallet-wallet-2"
  >
    <div
      className="Wallet-walletTitle-3"
    >
      <p>
        My Address:
      </p>
    </div>
    <div
      className="Wallet-walletAddress-5"
    >
      <img
        alt=""
        className="Wallet-copyIcon-4"
        onClick={[Function]}
        src="copy.svg"
      />
    </div>
    <div
      className="MuiInputBase-root-33 MuiInput-root-20 MuiInput-underline-24 Wallet-selectCurency-6"
      onClick={[Function]}
    >
      <div
        className="MuiSelect-root-13"
      >
        <div
          aria-haspopup="true"
          aria-pressed="false"
          className="Wallet-selectField-8"
          id="select-curency"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex={0}
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </div>
        <input
          id="curencyDropdownWallet"
          name="curency"
          type="hidden"
          value="DCB"
        />
        <i
          className="Wallet-curencyIcon-9"
        />
      </div>
    </div>
    <ul
      className="Wallet-walletNav-11"
    >
      <li>
        <div
          className="Wallet-walletNavItem-12"
          onClick={[Function]}
        >
          DCB
        </div>
      </li>
      <li>
        <div
          className="Wallet-walletNavItem-12"
          onClick={[Function]}
        >
          USD
        </div>
      </li>
      <li>
        <div
          className="Wallet-walletNavItem-12"
          onClick={[Function]}
        >
          EUR
        </div>
      </li>
      <li>
        <div
          className="Wallet-walletNavItem-12"
          onClick={[Function]}
        >
          CPT
        </div>
      </li>
    </ul>
  </div>
  <div
    className="Action-root-54"
  >
    <div
      className="Action-actionsSend-59"
    >
      <form
        className="Action-form-57"
        onSubmit={[Function]}
      >
        <span
          className="Action-sendTitle-60"
        >
          Send now
        </span>
        <div
          className="Action-wrSendField-61"
        >
          <label
            htmlFor="addressKey"
          >
            Address:
          </label>
          <input
            className="Action-sendAddressField-62"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Put address you want to send the money"
            type="text"
            value=""
          />
        </div>
        <div
          className="Action-wrSendField-61"
        >
          <label
            htmlFor="amount"
          >
            Amount:
          </label>
          <input
            className="Action-sendAmmountField-63"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            placeholder="Set ammount"
            type="text"
            value=""
          />
        </div>
        <button
          className="Action-sendBtn-65"
          type="submit"
        >
          Send money
        </button>
      </form>
    </div>
    <div
      className="Action-actionsInvoice-67"
    >
      <span
        className="Action-invoiceTitle-68"
      >
        Invoice
      </span>
      <div
        className="Action-wrSendField-61"
      >
        <label
          htmlFor="invoice"
        >
          Email:
        </label>
        <input
          className="Action-invoiceAmmountField-64"
          name="invoice"
          placeholder="Set ammount you need to receive"
          type="text"
        />
      </div>
      <button
        className="Action-sendBtn-65"
        type="submit"
      >
        Send Invoice
      </button>
    </div>
  </div>
</div>
`;
